### ISSUE TITLE
CocoaPods dependency declaration prevents upgrading past 0.9.4

### ISSUE THREAD

**Issue by zwaldowski** [realm/jazzy#1069]
9255d77 uses `~>` to specify CocoaPods 1.5, which prevents resolution with CocoaPods 1.6 and 1.7; both of the latter have compatibility fixes with Xcode 10.2. This leaves me stuck on Jazzy 0.9.4, which has compatibility issues of its own with Xcode 10.2.

**Comment by jpsim** [comment]
Do you foresee any downsides to just bumping up the version requirements for jazzy to point to the latest version of CocoaPods?

**Comment by zwaldowski** [comment]
I think the issue with that would be "pulling forward" someone to a newer version of CocoaPods they didn't expect when they were trying to update Jazzy only (speaking in terms of Bundler, I guess). Ideally, we'd declare the most generic version that supports the API we use, like with a `>=`. I don't use Jazzy's podspec support, so I don't have enough context to make a perfect recommendation here.

**Comment by johnfairh** [comment]
I tried to require cocoapods 1.6 as part of that jazzy release and hit problems to do with swift version handling meaning some podspecs would no longer build docs.  Should look at it soon, I guess will target 1.7 as that looks imminent and `supports_swift_versions` that might help.

I'm probably not understanding your problem, but Ruby is happy to have multiple versions of a gem installed at once: I have both latest cocoapods (1.6.1) and 1.5.3 installed so `pod --version` gives 1.6.1 and jazzy picks up the earlier (working) version via its Gemfile.  

(The cocoapods issue was, as best I recall, that in 1.6 it has a strong check that the swift version is set for each target.  This is good because our long-standing workaround to set the swift version via xcodebuild argument can go away.  This is bad because we need to do something in our internal Podfile construction to set a swift version in the case where some spec in the graph does not have one.  Which, er, ultimately will prolly be the same logic as the existing workaround.)

**Comment by zwaldowski** [comment]
Whether or not ruby can have multiple versions of a thing installed isn't relevant. Community best practices tell me to use a dependency manager. When I have weeks of investment into synchronizing developers' workspaces and CI environments, it's non-trivial to tell me to just opt out of it.

Either way, it doesn't make sense to have the project have a deleterious effect on build environments because of an optional feature of Jazzy.

**Comment by m-ruhl** [comment]
I have the same problem.. I updated my project to Swift 5 and CocoaPods 1.6
Therefore bundler only install jazzy 0.9.4

And this version does not support Swift 5:
```
❯ jazzy
Running xcodebuild
Could not parse compiler arguments from `xcodebuild` output.
Please confirm that `xcodebuild` is building a Swift module.
Saved `xcodebuild` log file: /var/folders/63/cp13lbv15csdqwx6mb84xsw40000gp/T/xcodebuild-459A0018-581F-485C-9588-300611D1C576.log
Failed to generate documentation
/Users/mr/.rvm/gems/ruby-2.3.0/gems/jazzy-0.9.5/lib/jazzy/executable.rb:36:in `execute_command': /Users/mr/.rvm/gems/ruby-2.3.0/gems/jazzy-0.9.5/bin/sourcekitten ["doc", "--"] (RuntimeError)

Running xcodebuild

Could not parse compiler arguments from `xcodebuild` output.

Please confirm that `xcodebuild` is building a Swift module.

Saved `xcodebuild` log file: /var/folders/63/cp13lbv15csdqwx6mb84xsw40000gp/T/xcodebuild-459A0018-581F-485C-9588-300611D1C576.log

Failed to generate documentation
```

**Comment by johnfairh** [comment]
@m-ruhl the stack trace there says jazzy 0.9.5; can you look at/share the Xcode build log to see what the problem being reported is?

**Comment by m-ruhl** [comment]
@johnfairh yeah.. saw this too.. If I run it now it uses the 0.9.4.

Here is the build log:
```
Build settings from command line:
    CODE_SIGN_IDENTITY = 
    CODE_SIGNING_REQUIRED = NO

note: Using new build system

** CLEAN SUCCEEDED **

note: Using new build system
note: Planning build
note: Constructing build description
Build system information
error: An empty identity is not valid when signing a binary for the product type 'Application'. (in target 'Scrum Poker')

** BUILD FAILED **
```

**Comment by johnfairh** [comment]
@m-ruhl that looks like #1057, try the workaround there.

**Comment by johnfairh** [comment]
Fixed in master via #1076 .
